src/helpers.ts

```diff
@@ -0,0 +1,14 @@
+type WithId = {
+  id: number
+};
+
+export const findById = <T extends WithId>(a: number) => {
+  return (b: T) => a === b.id;
+};
```

Comment:
You called the function `findById`, but you are not finding for anything here, but only comparing values. 
It is better to completely transfer the finding logic to the helper function